compiler/noirc_frontend/src/hir_def/traits.rs

```diff
@@ -11,6 +12,9 @@ pub struct TraitFunction {
     pub arguments: Vec<Type>,
     pub return_type: Type,
     pub span: Span,
+    pub default_impl: Option<Box<NoirFunction>>,
```

Comment:
Do we need an entire NoirFunction for the default_impl field? It seems like the only difference between a trait function with and without a default impl would be whether it has a body or not.